### ISSUE TITLE
gatsby-plugin-catch-links fails if pathPrefix exists

### ISSUE THREAD

**Issue by gera2ld** [gatsbyjs/gatsby#8907]
## Description

If pathPrefix exists, *gatsby-plugin-catch-links* works incorrectly for links in Markdown files. As a result, the prefix has been added twice, once by *gatsby-transformer-remark*, and once by `navigate`.

### Steps to reproduce

Demo project: https://github.com/gera2ld/demo-gatsby

```sh
# build with prefix `/public`
$ yarn build --prefix-paths

# start a local server
$ python3 -m http.server

# open http://localhost:8000/public/
```

### Expected result

Click on `page-2`, the expected behavior is to navigate to `/public/page-2/`.

### Actual result

Navigated to `/public/public/page-2/` and got 404.

### Environment

```
  System:
    OS: macOS High Sierra 10.13.6
    CPU: x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.3.0 - ~/.nvm/versions/node/v10.3.0/bin/node
    Yarn: 1.10.1 - /usr/local/bin/yarn
    npm: 6.1.0 - ~/.nvm/versions/node/v10.3.0/bin/npm
  Browsers:
    Chrome: 69.0.3497.100
    Firefox: 63.0
    Safari: 12.0
  npmPackages:
    gatsby: ^2.0.0 => 2.0.19
    gatsby-plugin-catch-links: ^2.0.4 => 2.0.4
    gatsby-plugin-manifest: ^2.0.2 => 2.0.4
    gatsby-plugin-offline: ^2.0.5 => 2.0.5
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.0
    gatsby-source-filesystem: ^2.0.2 => 2.0.2
    gatsby-transformer-remark: ^2.1.6 => 2.1.6
```

**Comment by JimTheMan** [comment]
same issue for me

**Comment by chasemccoy** [comment]
Has this change been published? I am still seeing this issue on the latest version of the plugin.

**Comment by pieh** [comment]
Published `gatsby-plugin-catch-links@2.0.5`

**Comment by pauleveritt** [comment]
2.0.10 appears to still have this issue.
